### PR TITLE
311 implement a prepare corpus command

### PR DESCRIPTION
**Pull request by yaseminbridges**
Implemented a `prepare-corpus` command.

A user can specify if they intend to carry out variant, gene, and/or disease analysis with their prepared corpus. The command will check the corpora and ensure that all required fields in the phenopacket are filled (required for matching results to known entities in the benchmarking). When a phenopacket is removed from the final corpus a warning message is logged, informing of the phenopacket that was removed.

There is also the option of spiking VCF with variant information from a phenopacket and updating the gene identifiers.

For example, to prepare the phenopacket-store corpus for our use (variant analysis only) we would use the command:

```bash
pheval-utils prepare-corpus --phenopacket-dir /path/to/phenopacket-store --gene-identifier ensembl_id --variant-analysis --template-vcf /path/to/template.vcf --output-dir /path/to/output_dir
```

This will:

1. Check that all vcf fields in the phenopacket are filled and remove those with missing values from the final corpus.
2. Update all the identifiers in the remaining phenopackets to ensembl
3. Spike variants into the template VCF

### CODE REVIEW

**Comment by yaseminbridges** [comment]
Hold off on merging this until #314 is checked and merged

**Comment by julesjacobsen** [comment]
Hi @yaseminbridges I pushed the 'convert to draft' link under the reviewers names in the 'reviewers' panel at the top right of this. If you're ready for review, press the 'Ready fro review' button in the merge checks box above this comment.

**Comment by yaseminbridges** [comment]
> Hold off on merging this until #314 is checked and merged



> Hi @yaseminbridges I pushed the 'convert to draft' link under the reviewers names in the 'reviewers' panel at the top right of this. If you're ready for review, press the 'Ready fro review' button in the merge checks box below this comment.

Hi @julesjacobsen, this is ready

File: src/pheval/cli_pheval_utils.py
Comment (julesjacobsen) [inline src/pheval/cli_pheval_utils.py:698]: For the `vcf_template` options it would be helpful to say that at least one of these are required for running a variant analysis or if they are for defining a custom user-defined VCF.

File: src/pheval/prepare/prepare_corpus.py
Comment (julesjacobsen) [inline src/pheval/prepare/prepare_corpus.py:33]: same as previous comment

File: src/pheval/utils/phenopacket_utils.py
Comment (julesjacobsen) [inline src/pheval/utils/phenopacket_utils.py:493]: Is this the case in the extended LIRICAL dataset? Isn't it just the case that the diagnosed variants should be empty?

File: tests/test_phenopacket_utils.py
Comment (julesjacobsen) [inline tests/test_phenopacket_utils.py:226]: Shouldn't this just be an empty list rather than a default null-type?

File: src/pheval/utils/phenopacket_utils.py
Comment (yaseminbridges) [reply src/pheval/utils/phenopacket_utils.py:493]: No, as it is still able to access the objects it returns these values when I access them in with Python, so still creates a list, just with empty string values

**Review by julesjacobsen — APPROVED** [review]

**Pull request merged**
